solr/core/src/test/org/apache/solr/search/RankQueryTest.java

```diff
@@ -106,4 +119,65 @@ public void testPluggableCollector() {
         "//result/doc[2]/str[@name='id'][.='6']",
         "//result/doc[1]/str[@name='id'][.='5']");
   }
+
+  // The following static classes are intended to ensure that support of covariant
+  // ScoreDocs is supported in rank queries. MyRankQuery will fail to compile
+  // if covariant ScoreDocs are not supported because it returns a TopDocsCollector
+  // for MyScoreDoc (a subtype of ScoreDoc).
+  static class MyScoreDoc extends ScoreDoc {
+    public int someOtherField = 0;
```

Comment:
```suggestion
    final public int someOtherField = 0;
```

Reply:
My bad, I meant `final public int someOtherField` here i.e. forgot to remove the ` = 0` when making the suggestion. https://github.com/apache/solr/pull/2506/commits/4bd9f94210e66ad1c0bd1d3abf5226fbc608307f to add the `final`.

Reply:
Oh, I thought you wanted it final because I didn't expose it originally in the constructor. It's fine in this test example, but in practice, it would need to remain mutable so that the `ScoreDoc` can be reused and changed within a `PriorityQueue` - none of the other fields in `ScoreDoc` are final.

Reply:
Ah, good point about it needing to be mutable in practice. I'd noticed it being public and there being no accessors and presumed that modification would/should be via an accessor and marking `final` would allow access without modification (presuming that no modification is needed after construction).

Good point about none of the other `ScoreDoc` fields being `final` -- so yes, let's keep things consistent here then and for `someOtherField` to be mutable too. Good catch!

Reply:
https://github.com/apache/solr/pull/2506/commits/c078d39a5cf127a1891f16ca0d3cf668b6a4596e reverts my earlier commit.